lib/tests/streamlit/server_test.py

```diff
@@ -593,3 +600,53 @@ def test_message_cache(self):
         # Cache misses
         self.assertEqual(404, self.fetch("/message").code)
         self.assertEqual(404, self.fetch("/message?id=non_existent").code)
+
+
+class ScriptCheckTest(tornado.testing.AsyncTestCase):
```

Comment:
@tconkling reagrarding the trivial tests, I find the `HealthHandlerTest` not relevant, since we are declaring in the test teh `/healthz`. What my tests should do, would be to check if a router for `/check-script-health` was added in the routers if the flag was True. I couldn't find any examples in the code were we actually test this, except maybe for the metrics endpoint, where the flag is inside the handler.

Since I reused the `HealthHandler`, I thought of the following tests:
1. Check that the router is created or not when I call `server._create_app` 
2. Use a separate handler and test it similarly like `HealthHandler`
3. Use (2) and add the flag inside the handler, like we do with the metrics.
4. Initialize the server with a `AsyncHttpTest` and call the endpoint, but then I won't need to test the `does_script_execute()` separately.

Do you have a suggestion on how I should test that?

Reply:
I suggest mocking the actual script-check logic in the endpoint (that is, just mock the `does_script_run_without_error` function to return a constant value, like the `HealthHandlerTest` does). Then, yeah - check what the return value is for the endpoint in the various situations:

- Config set to `False`: endpoint returns 404
- Config set to `True`, script-check succeeds: return 200
- Config `True`, script-check fails with timeout/error: return 503, "<expected message>"

Reply:
To be maybe a bit more concrete: there are two components to this PR, 

- the actual script-check logic (which is tested in your `ScriptCheckTest`)
- and the script-check endpoint, which is what I was requesting tests for here.
 
The endpoint tests don't need to test the script-check logic, they just need to test what the endpoint does when the script-check function returns with various values. (So, mocking the `does_script_run_without_error` in the endpoint tests might be useful for those tests)